test/unit/ajax.js

```diff
@@ -1408,6 +1408,103 @@ QUnit.module( "ajax", {
 		};
 	} );
 
+	ajaxTest( "jQuery.ajax() - responseURL", 13, function( assert ) {
+		function expectedUrl( url ) {
+			return jQuery.support.responseURL ? url : "";
+		}
+
+		/**
+		 * this function generates complete
+		 * absolute URL without any relative path components
+		 * example:
+		 * input: http://localhost:9876/base/test/data/testinit.js/../../../test/data/mock.php?action=responseURL&155821644568982930
+		 * output: http://localhost:9876/base/test/data/mock.php?action=responseURL&155821644568982930
+		 */
+		function removeRelativePathComponents( url ) {
```

Comment:
Why is this function necessary? We usually just use our existing `url` function.

Reply:
This is needed for the case when there is no redirect, in that case, `responseURL` should be equal to the initial URL only. The URL returned by `url` function contains some relative path components like `../../` (checkout the example input to this function in comments) so what happens is that after getting the response, `responseURL` value doesn't contain `../` and its cleaned because of which I am not able to assert it. help me with alternatives here

Reply:
hey @mgol can you suggest anything here?

Reply:
@csmadhav I’m at a conference; I’ll have a look next week unless I get to it before.

Reply:
OK, it seems the function can stay but:
1. I'd rename it to `normalizeUrl`; shorter & still fully conveys the meaning.
1. The implementation seems too complex for what it does, how about:

```js
function normalizeUrl( url ) {
	var finalUrl = url;
	while ( finalUrl.indexOf( '/../' ) > -1 ) {
		finalUrl = normalizedUrl.replace(/\/[^/]+\/\.\.\//g, '/');
	}
	return finalUrl;
}
```

Reply:
Yes, that would solve!